src/analysis/special_functions/stirling.lean

```diff
@@ -0,0 +1,237 @@
+/-
+Copyright (c) 2022. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Moritz Firsching, Fabian Kruse, Nikolas Kuhn
+-/
+import analysis.p_series
+import analysis.special_functions.log.deriv
+
+/-!
+# Stirling's formula
+
+This file proves Theorem 90 from the [100 Theorem List] <https://www.cs.ru.nl/~freek/100/>.
+It states that $n!$ grows asymptotically like $\sqrt{2\pi n}(\frac{n}{e})^n$.
```

Comment:
```suggestion
This file proves Stirling's formula for the factorial.
It states that $n!$ grows asymptotically like $\sqrt{2\pi n}(\frac{n}{e})^n$.
```
The fact that Stirling's formula is on that list is not so important that it should be mentioned in mathlib.